pkgs/games/rare/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib, fetchPypi, buildPythonApplication, makeDesktopItem, copyDesktopItems, qt5
+, pillow, psutil, pypresence, pyqt5, python, qtawesome, requests }:
+
+buildPythonApplication rec {
+  pname = "rare";
+  version = "1.6.2";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "Rare";
+    sha256 = "sha256-9xwqWXdw3BKieyJP/o2i9od6qbu5TVkYga8Q8cCMDcc=";
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    qt5.wrapQtAppsHook
+  ];
+
+  propagatedBuildInputs = [
+    pillow
+    psutil
+    pypresence
+    pyqt5
+    qtawesome
+    requests
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = "rare";
+      icon = "Rare";
+      comment = meta.description;
+      desktopName = "Rare";
+      genericName = "Rare (Epic Games Launcher Open Source Alternative)";
+    })
+  ];
+
+  dontWrapQtApps = true;
+
+  preBuild = ''
+    # Solves "PermissionError: [Errno 13] Permission denied: '/homeless-shelter'"
```

Comment:
What tries to write there? that is more important.

Reply:
Here is the full error message:

```
Traceback (most recent call last):
  File "/build/Rare-1.6.2/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 5, in <module>
    from rare import __version__ as version
  File "/build/Rare-1.6.2/rare/__init__.py", line 16, in <module>
    os.makedirs(cache_dir)
  File "/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```

Checking the [`__init__.py` file](https://github.com/Dummerle/Rare/blob/746aa671e05fec27b45f112645c2cef2f807b8db/rare/__init__.py#L10), as it hasn't found a cache directory set on `XDG_CACHE_HOME`, it tries to create a `~/.cache/rare` directory. Later, it also tries to find an `XDG_DATA_HOME`, but as there is nothing set, then it also tries to create a `~/.local/share/rare/` directory.